[13.0][IMP+FIX] account_asset_management: Add chatter entry in invoice for generated asset, FIX reference empty

ValentinVinagre opened the pull request:
**IMP**:
Coming from: https://github.com/OCA/account-financial-tools/issues/1006
Added chatter entry with the link to the asset.

**FIX** (empty asset reference):
If the assets are created before posting the invoice, they wouldn't have a name and the "code" field of the assets will be empty, but they would have the invoice name.

ValentinVinagre committed:
[IMP] account_asset_management: Add chatter entry in invoice for generated asset

ValentinVinagre commented:
@HaraldPanten could you review it?
Thanks

pedrobaeza commented on account_asset_management/models/account_move.py:
```diff
@@ -59,6 +59,7 @@ def write(self, vals):
         return super().write(vals)
 
     def post(self):
+        super().post()
```
Why this change?

ValentinVinagre replied:
Because if the assets are created before posting the invoice, they wouldn't have a name and the "code" field of the assets will be empty, but they would have the invoice name.

pedrobaeza replied:
But that is another thing. Please always split these things is separate PRs for having separate debate. I'm also suspecting that the culprit of that is our module `l10n_es_account_invoice_sequence`, not this module.

HaraldPanten replied:
> But that is another thing. Please always split these things is separate PRs for having separate debate. I'm also suspecting that the culprit of that is our module `l10n_es_account_invoice_sequence`, not this module.

I don't think so. As we are working for V13 and l10n_es_account_invoice_sequence does not exist for V13. Might you be talking about V12?

pedrobaeza replied:
Yes, I was talking in v12, that the operation is performed that way. Then it can be of this module, but please put a separate PR with this fix (that doesn't affect the chatter entry), and explain in detail the problem in bot the PR and the commit (`FIX reference empty`) is not enough explanation.

HaraldPanten replied:
Just to know... Would it be OK to create a single PR but several commits for different FIXs and IMPs as long as each was well explained?

pedrobaeza replied:
That can serve, but take into account that if one of the changes generate debate, it will delay the merge of the other features that can be very clear, so you have to balance this for deciding if putting one PR or several.

ValentinVinagre replied:
@pedrobaeza as the IMP depends on the FIX, can I do 2 commits?

pedrobaeza replied:
I don't think they depend each other, but you get a "better" chatter entry when the FIX will arrive, but OK.

ValentinVinagre replied:
Done

HaraldPanten reviewed (approved):
It LGTM 👍

ValentinVinagre committed:
[FIX] account_asset_management: FIX empty asset reference

If the assets are created before posting the invoice, they wouldn't have a name and the "code" field of the assets will be empty, but they would have the invoice name.

pedrobaeza commented:
You are still explaining nothing in the commit message, and the super was in the middle by a reason. @luc-demeyer can you tell us why it was that way?

ValentinVinagre commented:
> You are still explaining nothing in the commit message, and the super was in the middle by a reason. @luc-demeyer can you tell us why it was that way?

I changed the last commit and the header of the PR.

HaraldPanten reviewed (approved):
LGTM 👍

luc-demeyer commented on account_asset_management/models/account_move.py:
```diff
@@ -79,7 +80,15 @@ def post(self):
+                    "asset_profile_id"
+                ).asset_id.name_get()
+            ]
+            message = _("This invoice created the asset/s: %s") % ", ".join(refs)
```
I think we should change 'asset/s' into 'asset(s)'

ValentinVinagre replied:
Done!

pedrobaeza commented:
/ocabot merge patch
